### ISSUE TITLE
Centalized logging (for Cloud)

### ISSUE THREAD

**Issue by dan-mckean** [sourcegraph/sourcegraph#21598]
`Security overflow work `

Originally from [security work for cloud doc](https://docs.google.com/document/d/1rZsHOr0CH1RP5ahsG_YFVJLF_7uhtgK93Fuh77SD2gE/edit#bookmark=id.v259t66sbncz)

## Centralized logging
~~Tune & configure elastic cloud~~
Configure GCP Cloud Logging

Success criteria
 * We are confident that we know what questions we want to ask our logs
 * We are confident that we can answer these questions with the data we collect
 * We are confident that we're not ingesting too many logs that are irrelevant to security
 * Note that filtering out logs is a dangerous game, so we'll err on the side of caution when filtering things out
 * We are confident we can identify the scope of a breach of cloud by investigating our logs
 ~~We are confident that Elastic cloud is stable for the foreseeable future~~
 * We are confident that our logs will be retained for 12 (negotiable) months

Bonus success criteria
 * We have a staging environment in which to test changes we apply to our production logging instance

**Comment by daxmc99** [comment]
The distribution team would prefer to use GCP logging for this, any objections from @sourcegraph/security ? 
This would eliminate the need to tune or scale elastic.

**Comment by ElizabethStirling** [comment]
I'm fine with this, as long as we can query across logs & retain the option to push things to Elastic later if needed. cc @andreeleuterio  - thoughts?

**Comment by daxmc99** [comment]
Current outline: 

Directing all logs from Sourcegraph Cloud folder to separate security logging project. 

Currently writing the terraform to accomplish this

**Comment by daxmc99** [comment]
See some previous work here

**Comment by daxmc99** [comment]
Current status:
EOD Terraform PR to add logging buckets, setup IAM policy.

**Comment by daxmc99** [comment]
External logs from Cloudflare are currently blocked on having an enterprise account. https://developers.cloudflare.com/logs/

**Comment by ElizabethStirling** [comment]
We've recently upgraded to enterprise : https://sourcegraph.slack.com/archives/C01UYJAET9P/p1622560467027500

**Comment by daxmc99** [comment]
Cloudflare logs are also being ingested into our cloudflare audit bucket. 

We now have centralized logs setup here: https://console.cloud.google.com/storage/browser?organizationId=1006954638239&project=sourcegraph-security-logging&prefix=

**Comment by daxmc99** [comment]
Closing this issue as this work is done. Creating this other issue to track the stretch goal of gathering audit logs for all managed instances. https://github.com/sourcegraph/sourcegraph/issues/22422
